src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,25 @@
+package core.basesyntax;
+
+public class Main {
+    public static void main(String[] args) {
+        int arraySize = 6; // array size
+        Figure[] figures = new Figure[arraySize];
+        FigureSupplier figureSupplier = new FigureSupplier();
+
+        for (int i = 0; i < arraySize; i++) {
+            if (i < figures.length / 2) {
+                Figure randomFigure = figureSupplier.getRandomFigure();
+                figures[i] = randomFigure;
+            } else {
+                Figure defaultFigure = figureSupplier.getDefaultFigure();
+                figures[i] = defaultFigure;
```

Comment:
In situations like this, there is no need to store the object in a local variable if it will be immediately used. Instead, you can pass the expression directly without assigning it to a variable